extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-postgresql/schema/upgrade/upgrade-pre-1.6.0.sql

```diff
@@ -0,0 +1,26 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one
+-- or more contributor license agreements.  See the NOTICE file
+-- distributed with this work for additional information
+-- regarding copyright ownership.  The ASF licenses this file
+-- to you under the Apache License, Version 2.0 (the
+-- "License"); you may not use this file except in compliance
+-- with the License.  You may obtain a copy of the License at
+--
+--   http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing,
+-- software distributed under the License is distributed on an
+-- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+-- KIND, either express or implied.  See the License for the
+-- specific language governing permissions and limitations
+-- under the License.
+--
+
+--
+-- Add new system-level audit permission
+--
+
+ALTER TYPE guacamole_system_permission_type
+    ADD VALUE 'AUDIT'
+    BEFORE 'ADMINISTER';
```

Comment:
Why before `ADMINISTER`? Alphabetically it should go after...

Reply:
Yeah, that's a fair point - I was probably thinking privilege level vs. alphabetic, but I can re-organize it to make sure it's consistent.

Reply:
Actually, looking at the other code, it's because that's how those enums tend to be organized:

https://github.com/apache/guacamole-client/blob/6493a2313bd976440fea9479742715040662ed74/extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-postgresql/schema/001-create-schema.sql#L53-L60

`ADMINISTER` is the last item in that list, and, when I added `AUDIT` to the creation script, I left it that way, so when I do the `ALTER TYPE`, here, in the upgrade script, I insert it in the same way that it would be for the creation script.

Reply:
Fair

Reply:
Now, that said, what I'm not sure is if the order it's added in has any implications for how the database stores and/or (re)indexes the existing values. So, if I add it before, does Postgres (in this case) have to go through and update references to it in tables that have that permission? Whereas if I just add it after, it becomes the next value and no updates need to be done?

Reply:
Hi @jmuehlner and @mike-jumper, thanks for moving this topic.
Is there something that I can help with?